src/Synth/Envelope.cpp

```diff
@@ -187,11 +216,12 @@ float Envelope::envout(bool doWatch)
 
         return out;
     }
-    if(inct >= 1.0f)
+
+    if(inct >= 1.0f) // if we reached the next point
         out = envval[currentpoint];
-    else
-        out = envval[currentpoint - 1]
-              + (envval[currentpoint] - envval[currentpoint - 1]) * t;
+    else { // if we have to interpolate between points
```

Comment:
Am I correct in thinking this PR removes linear mode?

Reply:
> Am I correct in thinking this PR removes linear mode?

if (envcpval == 0) the curve is linear

Reply:
@fundamental Does the answer solve your concern?